libs/parallelism/algorithms/include/hpx/parallel/algorithms/uninitialized_fill.hpp

```diff
@@ -258,65 +406,141 @@ namespace hpx { namespace parallel { inline namespace v1 {
+#endif
+        return detail::uninitialized_fill_n<FwdIter>().call(
             std::forward<ExPolicy>(policy), first, std::size_t(count), value);
+#if defined(HPX_GCC_VERSION) && HPX_GCC_VERSION >= 100000
+#pragma GCC diagnostic pop
+#endif
     }
 }}}    // namespace hpx::parallel::v1
+
+namespace hpx {
+    ///////////////////////////////////////////////////////////////////////////
+    // DPO for hpx::uninitialized_fill
+    HPX_INLINE_CONSTEXPR_VARIABLE struct uninitialized_fill_t final
+      : hpx::functional::tag_fallback<uninitialized_fill_t>
+    {
+        // clang-format off
+        template <typename FwdIter, typename T,
+            HPX_CONCEPT_REQUIRES_(
+                hpx::traits::is_forward_iterator<FwdIter>::value
+            )>
+        // clang-format on
+        friend void tag_fallback_dispatch(hpx::uninitialized_fill_t,
+            FwdIter first, FwdIter last, T const& value)
+        {
+            static_assert(hpx::traits::is_forward_iterator<FwdIter>::value,
+                "Requires at least forward iterator.");
+
+            hpx::parallel::v1::detail::uninitialized_fill<FwdIter>().call(
+                hpx::execution::seq, first, last, value);
+        }
+
+        // clang-format off
+        template <typename ExPolicy, typename FwdIter, typename T,
+            HPX_CONCEPT_REQUIRES_(
+                hpx::is_execution_policy<ExPolicy>::value &&
+                hpx::traits::is_forward_iterator<FwdIter>::value
+            )>
+        // clang-format on
+        friend typename parallel::util::detail::algorithm_result<ExPolicy>::type
+        tag_fallback_dispatch(hpx::uninitialized_fill_t, ExPolicy&& policy,
+            FwdIter first, FwdIter last, T const& value)
+        {
+            static_assert(hpx::traits::is_forward_iterator<FwdIter>::value,
+                "Requires at least forward iterator.");
+
+            using result_type =
+                typename hpx::parallel::util::detail::algorithm_result<
+                    ExPolicy>::type;
+
+            return hpx::util::void_guard<result_type>(),
+                   hpx::parallel::v1::detail::uninitialized_fill<FwdIter>()
+                       .call(
+                           std::forward<ExPolicy>(policy), first, last, value);
+        }
+
+    } uninitialized_fill{};
+
+    ///////////////////////////////////////////////////////////////////////////
+    // DPO for hpx::uninitialized_fill_n
+    HPX_INLINE_CONSTEXPR_VARIABLE struct uninitialized_fill_n_t final
+      : hpx::functional::tag_fallback<uninitialized_fill_n_t>
+    {
+        // clang-format off
+        template <typename FwdIter, typename Size, typename T,
+            HPX_CONCEPT_REQUIRES_(
+                hpx::traits::is_forward_iterator<FwdIter>::value
+            )>
+        // clang-format on
+        friend FwdIter tag_fallback_dispatch(hpx::uninitialized_fill_n_t,
+            FwdIter first, Size count, T const& value)
```

Comment:
Should we add a `std::is_integral<Size>::value` constraint here?

Reply:
Oh ok sure, I'll add that

Reply:
Should I add the same for uninitialized_copy_n and uninitialized_move_n as well?

Reply:
If applicable, sure.